agent/03_agent_deployment.sh

```diff
@@ -107,17 +134,64 @@ stringData:
   .dockerconfigjson: '${pull_secret}'
 
 EOF
+
+    ip_list=$2
+
+    num_ips=${#ip_list[@]}
+
+    # Create a yaml for each host in nmstateconfig.yaml
+    for (( i=0; i<$num_ips; i++ ))
+    do
+        cat >> "${FLEETING_MANIFESTS_PATH}/nmstateconfig.yaml" << EOF
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  name: mynmstateconfig
+  namespace: openshift-machine-api
+  labels:
+    cluster0-nmstate-label-name: cluster0-nmstate-label-value
+spec:
+  config:
+    interfaces:
+      - name: eth0
+        type: ethernet
+        state: up
+        mac-address: ${mac_list[i]}
+        ipv4:
+          enabled: true
+          address:
+            - ip: ${ip_list[i]}
+              prefix-length: ${CLUSTER_HOST_PREFIX}
+          dhcp: false
+    dns-resolver:
+      config:
+        server:
+          - ${PROVISIONING_HOST_EXTERNAL_IP}
+    routes:
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: ${PROVISIONING_HOST_EXTERNAL_IP}
+          next-hop-interface: eth0
+          table-id: 254
+  interfaces:
+    - name: "eth0"
+      macAddress: ${mac_list[i]}
+---
+EOF
+    done
+
     set -x
 }
 
 function generate_fleeting_iso() {
     export REPO_PATH=${WORKING_DIR}
+    ip_list=$1
+
     sync_repo_and_patch fleeting https://github.com/openshift-agent-team/fleeting ${FLEETING_PR}
 
-    generate_fleeting_manifests ${FLEETING_PATH}
-    
+    generate_fleeting_manifests ${FLEETING_PATH} $ip_list $2 $3
```

Comment:
I'm having trouble testing it. Seeing this error:
`./agent/03_agent_deployment.sh: line 192: $2: unbound variable`

Reply:
Thanks @rwsu, that should be fixed with latest update.